pkg/runtime/serializer/versioning/versioning.go

```diff
@@ -102,11 +117,30 @@ func (c *codec) Decode(data []byte, defaultGVK *unversioned.GroupVersionKind, in
 			}
 			return into, gvk, nil
 		}
+
+		// perform defaulting if requested
+		if c.defaulter != nil {
+			// create a copy to ensure defaulting is not applied to the original versioned objects
```

Comment:
The contract of "VersionedObjects" is that we return the decoded, unmodified external version, and the internal, modified version.  In the old code (conversion+defaulting) only the second object had defaults applied.  This is preserving that behavior (various tests caught it).